assets/js/controllers/edit_controller.js

```diff
@@ -0,0 +1,85 @@
+/*!
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+  static targets = ['tab', 'tabStore'];
+
+  connect() {
+    if (this.tabSelected) {
+      this.showFirstTabWithErrors('.sonata-ba-field-error');
+    }
+  }
+
+  checkValidity() {
+    if (this.tabSelected) {
+      this.showFirstTabWithErrors(':invalid');
+    }
+  }
+
+  prepareSubmit() {
+    setTimeout(() => {
+      this.submitters.forEach((submitter) => {
+        submitter.disabled = true;
+      });
+    }, 1);
+
+    if (this.tabSelected) {
+      this.tabStoreTarget.value = this.tabSelected.getAttribute('aria-controls');
+    }
+  }
+
+  showFirstTabWithErrors(errorSelector) {
+    let firstTabWithErrors = null;
+
+    this.tabTargets.forEach((tab) => {
+      const pane = this.element.querySelector(tab.getAttribute('href'));
+      const icon = tab.querySelector('.has-errors');
+
+      if (pane.querySelectorAll(errorSelector).length > 0) {
+        // Only show first tab with errors
+        if (!firstTabWithErrors) {
+          jQuery(tab).tab('show');
```

Comment:
In 5.x branch will need update to `Tab.getOrCreateInstance(tab)`

Reply:
I'm sticking to the plan, that in branch 5.x we will update `bootstrap` to 5 version  and remove `jquery` from dependencies

Reply:
Oh ok